### PR TITLE
Simplified and updated rubocop rules #trivial

### PR DESCRIPTION
**Pull request by KrauseFx**
Simplified and updated rubocop rules #trivial

### CODE REVIEW

**Comment by KrauseFx** [comment]
cc @neonichu @orta 👍

File: lib/danger/plugin_support/plugin_parser.rb [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
+#       usable_methods = methods.select { |m| m.visibility == :public }.reject { |m| m.name == :initialize }
+#
+#
+#   the alternative, is to add
+#
+#       require 'pry'
+#       binding.pry
+#
+#   anywhere inside the source code below.
+#
```
Comment (orta) [inline lib/danger/plugin_support/plugin_parser.rb]: This change isn't wanted, you can't copy and paste these examples anymore

File: lib/danger/plugin_support/plugin_parser.rb [hunk 2]
```diff
@@ -73,6 +71,7 @@ def to_json
       to_dict(plugins).to_json
     end
 
+    # rubocop:disable Metrics/AbcSize
```
Comment (orta) [inline lib/danger/plugin_support/plugin_parser.rb]: 👍  great

**Comment by KrauseFx** [comment]
Updated the PR 🚀

**Comment by KrauseFx** [comment]
👍

File: .rubocop.yml [hunk 3]
```diff
@@ -108,19 +70,40 @@ Metrics/LineLength:
...
   Max: 18
 
-Style/DotPosition:
+# Sometimes it's easier to read without guards
+Style/GuardClause:
   Enabled: false
 
-Style/GuardClause:
+# We allow both " and '
+Style/StringLiterals:
```
Comment (DanToml) [inline .rubocop.yml]: We want this to be enabled for consistency see https://github.com/danger/danger/issues/300

**Comment by orta** [comment]
Woooooo

File: .rubocop_todo.yml [hunk 4]
```diff
@@ -1,15 +0,0 @@
-# This configuration was generated by
-# `rubocop --auto-gen-config`
-# on 2016-07-04 18:33:07 -0400 using RuboCop version 0.38.0.
-# The point is for the user to remove these configuration records
-# one by one as the offenses are removed from the code base.
-# Note that changes in the inspected code, or installation of new
-# versions of RuboCop, may require this file to be generated again.
-#
-
-Style/StringLiterals:
```
Comment (DanToml) [inline .rubocop_todo.yml]: We need these to enable StringLiteral checking. (On CI, Rubocop doesn't detect emoji in strings as being special characters, but seems to locally :/)

**Comment by DanToml** [comment]
@orta this just reverted #301 / #300 essentially too

**Comment by orta** [comment]
aww

**Comment by KrauseFx** [comment]
The reason why we should allow both `"` and `'` is to make it easier for non-Ruby developers. The difference between those 2 is not clear, and many people don't know that `#{...}` only works in combination with `"`, I'd also be in favour of `"` for this reason, as there is almost no performance difference between those 2.

**Comment by orta** [comment]
I'd totally be on board with converting to always be double quotes

**Comment by DanToml** [comment]
I'd be fine with having `"` everywhere. The perf difference in Danger won't be noticeable either way. But a nonsensical mix is more confusing to Non-Ruby developers. (because there's no rule at that point)

**Comment by KrauseFx** [comment]
+1 on the `"` always, I should do the same for the `fastlane` code base

**Pull request merged**